freqtrade/rpc/api_server/api_auth.py

```diff
@@ -35,15 +40,47 @@ def get_user_from_token(token, secret_key: str, token_type: str = "access"):
         payload = jwt.decode(token, secret_key, algorithms=[ALGORITHM])
         username: str = payload.get("identity", {}).get('u')
         if username is None:
-            raise credentials_exception
+            if raise_on_error:
+                raise credentials_exception
+            else:
+                return False
         if payload.get("type") != token_type:
-            raise credentials_exception
+            if raise_on_error:
+                raise credentials_exception
+            else:
+                return False
 
     except jwt.PyJWTError:
-        raise credentials_exception
+        if raise_on_error:
+            raise credentials_exception
+        else:
+            return False
     return username
 
```

Comment:
i think a MUCH simpler way (and also a lot less error-prone) for this is to accept the exception - and catch it in `get_ws_token`.

the exception IS correct - and will only be risen if auth is not there.
sure, it's slightly more expensive than an IF - but it'll only happen in "not authenticated" cases - which will not be the main usecase (we assume and recommend to never expose the freqtrade API directly on the internet).